File: 4. User input/main.cpp
#include <iostream>

using namespace std;

int main(){
    cout << "Please input your name: ";//First ask for user input
    string input;//First declare a string variable
    cin >> input;//Reads input from user then puts it in "input"
    cout << "Hello " << input << endl;//Outputs user input

    cout << "Please input a number: ";//Ask user to input number
    int number;//Declare an int variable
    cin >> number;//Puts user input into "number"
    cout << "You have entered: " << number;//Outputs second user input.

    //You need to declare a variable before you can use cin to assign user input to the variable
    //cin will automatically convert the input into whatever type is being inputted
}

File: 4. User input/main.cs
using System;

namespace DefaultNamespace
{
	public class Class1
	{
		void main(){
			Console.WriteLine ("Please input your name: ");//First ask for user input
			string input = Console.ReadLine();//Reads input from user then puts it in a string

			Console.WriteLine("Hello " + input);//Outputs user input
			Console.WriteLine("Please input a number: ");//Ask user to input number
			int number = Convert.ToInt32(Console.ReadLine);//Reads input from user then puts it into an Int, but needs explicit conversion first.
			Console.WriteLine("You have entered: " + number)//Output second user input.

			//Unlike C++, you can just assign Console.ReadLine() as if it was a variable.
			//But Console.ReadLine() returns as a String, so it needs conversion before being put into other types.
		}
	}
}
	
